runtime/jvmti/jvmtiStartup.c

```diff
@@ -311,9 +339,9 @@ criuRestoreStartAgent(J9JavaVM *vm)
 {
 	J9VMInitArgs *criuRestoreArgsList = vm->checkpointState.restoreArgsList;
 
-	processAgentLibraryFromArgsList(vm, criuRestoreArgsList, VMOPT_AGENTLIB_COLON, TRUE, FALSE);
-	processAgentLibraryFromArgsList(vm, criuRestoreArgsList, VMOPT_AGENTPATH_COLON, TRUE, FALSE);
-	processAgentLibraryFromArgsList(vm, criuRestoreArgsList, MAPOPT_XRUNJDWP, TRUE, TRUE);
+	processAgentLibraryFromArgsList(vm, criuRestoreArgsList, TRUE, FALSE, FALSE);
+	processAgentLibraryFromArgsList(vm, criuRestoreArgsList, TRUE, FALSE, TRUE);
+	processAgentLibraryFromArgsList(vm, criuRestoreArgsList, TRUE, TRUE, FALSE);
```

Comment:
I think these options should be processed in command-line order.

Reply:
Could you elaborate on the command-line order?

Reply:
I was expecting that `processAgentLibraryFromArgsList()` would scan arguments left to right looking for `-agentpath:`, `-agentlib:` or `-Xrunjdwp` in a _single_ pass rather than scan the whole argument list for `-agentpath:`, then scan the list again for `-agentlib:`, followed by a third scan looking for `-Xrunjdwp`. Or is there something I'm missing that makes the the order irrelevant?

Reply:
This helper method was to replace the duplicate code block: https://github.com/eclipse-openj9/openj9/blob/290c2edabc7ed4029ed9c0ca4d6180e74b2a89a4/runtime/jvmti/jvmtiStartup.c#L139-L211
the whole argument list was scanned for each option since each option could appear multiple times, i.e., multiple `-agentpath` or `-agentlib`. 
`-Xrunjdwp:` could appear multiple times as well but there is no need to search for a second one or more, added `break` after the first one is found.

Reply:
I understand that each of those options may appear multiple times, my concern is that the order of, say, one `-agentlib` option relative to a separate `-agentpath` or `-Xrunjdwp` is significant. Processing all `-agentlib` options separately from all `-agentpath` or `-Xrunjdwp` options ignores that fact. Is there significance to that order? If not, please explain why it doesn't matter, if it matters, please correct the code to respect that order.

Reply:
Existing code processes all `-agentlib` options separately from all other options, similarly for `-agentpath`, which assumes no significance to that order. There is little specification doc supporting this assumption or otherwise.
@gacholio any insights?

Reply:
Created 
* https://github.com/eclipse-openj9/openj9/issues/20259
for further investigation